Modules/_ctypes/cfield.c

```diff
@@ -144,11 +129,69 @@ PyCField_new_impl(PyTypeObject *type, PyObject *name, PyObject *proto,
                              ((PyTypeObject*)proto)->tp_name);
                 goto error;
         }
+
+        if (byte_size > 100) {
+            // Bitfields must "live" in a field defined by a ffi type,
+            // so they're limited to about 8 bytes.
+            // This check is here to avoid overflow in later checks.
+            PyErr_Format(PyExc_ValueError,
+                        "bit field %R size too large, got %zd",
+                        name, byte_size);
+        }
+        bitfield_size = PyLong_AsSsize_t(bit_size_obj);
```

Comment:
Note for myself: because `bit_size_obj` should be small, we would use a fast path so this call would be efficient. However, if it's too large, then it will be slower, but we don't care as we will raise an exception.